openvpn/middlewares/client/bytescount/selective_stats_handler.go

```diff
@@ -0,0 +1,22 @@
+package bytescount
+
+import (
+	"github.com/pkg/errors"
+	"time"
+)
+
+// NewIntervalStatsHandler creates and returns composite handler, which invokes internal handler at given interval
+func NewIntervalStatsHandler(handler SessionStatsHandler, clock func() time.Time, interval time.Duration) (SessionStatsHandler, error) {
+	if interval < 0 {
+		return nil, errors.New("Invalid 'interval' parameter")
+	}
+	var lastTime *time.Time = nil
+	return func(sessionStats SessionStats) error {
+		now := clock()
+		if lastTime == nil || (now.Sub(*lastTime)) >= interval {
```

Comment:
Maybe lastTime.Add(interval) < now ?

Reply:
Not sure if this is better, especially since you've made a mistake in your example: `>=` should be used instead of `<` :D

Reply:
No I didn't.

Reply:
Yeah, you didn't - it's more confusing than I initially thought :D
Comparing passed time with interval time seems much easier.